frontend/src/components/Header.jsx

```diff
@@ -3,26 +3,102 @@ import ButtonVariant from './ButtonVariant'
 import ProfileButton from './ProfileButton'
 import SearchBar from './SearchBar'
 import { Link } from 'react-router-dom'
-import { useWeb3React } from '@web3-react/core'
+import { useWeb3React, UnsupportedChainIdError } from '@web3-react/core'
 import { injected } from '../utils/connectors'
+import {
+    NoEthereumProviderError,
+    UserRejectedRequestError
+  } from '@web3-react/injected-connector';
+  import { useState } from 'react';
 
+function getErrorMessage(error) {
+    let errorMessage;
+  
+    switch (error.constructor) {
+      case NoEthereumProviderError:
+        errorMessage = `No Ethereum browser extension detected. Please install MetaMask extension.`;
+        break;
+      case UnsupportedChainIdError:
+        errorMessage = `You're connected to an unsupported network.`;
+        break;
+      case UserRejectedRequestError:
+        errorMessage = `Please authorize this website to access your Ethereum account.`;
+        break;
+      default:
+        errorMessage = error.message;
+    }
+  
+    return errorMessage;
+  }
+
+function Activate(){
+    const context = useWeb3React();
+    const { activate, active } = context;
+    const [activating, setActivating] = useState(false);
+
+    function handleActivate(event){
+        event.preventDefault();
+        async function _activate(activate) {
```

Comment:
LGTM overall. Here we don't need to await for the async activate fucntion since we are relying on a user input that's happening outside of our application. We do need to properly handle any errors like activate(injected, (err) =>{...})